src/geophires_x/CylindricalReservoir.py

```diff
@@ -244,20 +244,20 @@ def Calculate(self, model: Model) -> None:
                                                  ) / 1e15  # 10^15 J
         self.cpwater.value = heat_capacity_water_J_per_kg_per_K(
             model.wellbores.Tinj.value * 0.5 + (self.Trock.value * 0.9 + model.wellbores.Tinj.value * 0.1) * 0.5,
-            pressure=model.reserv.lithostatic_pressure()
+            pressure=self.lithostatic_pressure(model.reserv.rhorock.value, model.reserv.Trock.value)
         )
         self.rhowater.value = density_water_kg_per_m3(
             model.wellbores.Tinj.value * 0.5 + (self.Trock.value * 0.9 + model.wellbores.Tinj.value * 0.1) * 0.5,
-            pressure=model.reserv.lithostatic_pressure()
+            pressure=self.lithostatic_pressure(model.reserv.rhorock.value, model.reserv.Trock.value)
         )
 
         model.logger.info(f'complete {str(__class__)}: {sys._getframe().f_code.co_name}')
 
-    def lithostatic_pressure(self) -> PlainQuantity:
+    #def lithostatic_pressure(self) -> PlainQuantity:
         """
         @override
 
         Standard reservoir implementation uses depth but CylindricalReservoir sets depth to total drilled length
         """
-        return quantity(static_pressure_MPa(self.rhorock.quantity().to('kg/m**3').magnitude,
-                                            self.InputDepth.quantity().to('m').magnitude), 'MPa')
+        def lithostatic_pressure(self, rho_rock: float, depth: float) -> PlainQuantity:
+            return quantity(static_pressure_MPa(rho_rock, depth), 'MPa')
```

Comment:
The purpose of defining `lithostatic_pressure` as a member of self is to avoid having to pass arguments to it, so this change to it is redundant/unwanted as far as I can tell? If there are instances where it should be calculated with parameters other than `self.rhorock` and self depth, then those calls should be made to `static_pressure_MPa` with parameters passed as quantities in order to preserve units.

Reply:
I feel that "lithostatic_pressure" should be a function independent of context—you pass some values to it, and it passes back the answer. What is the purpose of making it context-dependent (i.e., it must be called within the Reservoir object to access the values it needs to calculate its result)? We need to be able to call it from other objects (as I needed to do with WellBores) with the arguments I needed. I can see the argument if a function can only be called from within one object but not multiple objects. In that case, I suggest lithostatic_pressure and any other such function should be in a shared utility space, like GeophiresUtils.py.